### ISSUE TITLE
Possible typo in the "Transfer learning & fine-tuning" guide

### ISSUE THREAD

**Issue by miticollo** [keras-team/keras#20285]
I'm not sure if this is issue. For now it is an observation.

Reading the guide about [Transfer learning & fine-tuning](https://github.com/keras-team/keras/blob/master/guides/transfer_learning.py) I had some trouble to understand the following statement: https://github.com/keras-team/keras/blob/38db71acb207fb1dda380a40df68636e45d46174/guides/transfer_learning.py#L234-L237

`base_model` is part of a Functional model that is trained with the `fit` method. It invokes the `call` method of the model with `training=True`: https://github.com/keras-team/keras/blob/38db71acb207fb1dda380a40df68636e45d46174/keras/src/backend/tensorflow/trainer.py#L51

This overrides the previous `training=False`. I tested it using [pdb](https://docs.python.org/3/library/pdb.html) and Google Colab. [Here](https://colab.research.google.com/drive/1gk383gklxOdUKUZsfn6eJm53UsCrkyBa?usp=sharing) the IPython notebook. As you can see in the last cell of the notebook `training` is `True` but the `BatchNormalization` layer (included in the Xception model) works in **inference mode** because `self.trainable` is `False`: https://github.com/keras-team/keras/blob/38db71acb207fb1dda380a40df68636e45d46174/keras/src/layers/normalization/batch_normalization.py#L254-L266

But after unfreezing the `base_model`: https://github.com/keras-team/keras/blob/38db71acb207fb1dda380a40df68636e45d46174/guides/transfer_learning.py#L531-L536

`BatchNormalization` layer works in **training mode**.

**Comment by miticollo** [comment]
I'm agree with you! 

**But `variance` and `mean` in `BatchNormalization` must be updated when the `base_model` is unfrozen?** Based on the following statements, **no**.

https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/guides/transfer_learning.py#L317-L321

https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/guides/transfer_learning.py#L470-L472

https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/guides/transfer_learning.py#L531-L535

I mean the following not happen: https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/keras/src/layers/normalization/batch_normalization.py#L34-L37
because `BatchNormalization` continues to train moving average of the mean and standard deviation.

**Comment by mehtamansi29** [comment]
Hi @miticollo -

> **But `variance` and `mean` in `BatchNormalization` must be updated when the `base_model` is unfrozen?** Based on the following statements, **no**.

Here on this line [keras/guides/transfer_learning.py](https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/guides/transfer_learning.py#L317-L321) Lines 317 to 321, BatchNormalization description is not updated. 

But here in file there is an important not regarding BatchNormalization layer[keras/guides/transfer_learning.py](https://github.com/keras-team/keras/blob/577ef63690ecc1326d4a69888c392228161ab484/guides/transfer_learning.py#L304-L321).
- `BatchNormalization` contains 2 non-trainable weights that get updated during training. These are the variables tracking the mean and variance of the inputs. 
- When you unfreeze a model that contains `BatchNormalization` layers in order to do fine-tuning, you should keep the `BatchNormalization` layers in inference mode by passing `training=False` when calling the base model. Otherwise the updates applied to the non-trainable weights will suddenly destroy what the model has learned.

More details regarding BatchNormalization in transfer learning you can find here in understanding the [trainable attribute ](https://keras.io/guides/transfer_learning/#introduction)section.

**Comment by miticollo** [comment]
Sorry, but I don't understand your answer. Maybe an example can be useful. Look [this]() IPython notebook.

As you can see at the end of the notebook `BatchNormalization` layers work in training mode even if `training=False` is set in `base_model`'s `call` method.

So, IMO, there is a typo in the guide or a bug in Keras.

**Comment by mehtamansi29** [comment]
Hi @miticollo -

I catch your point. I ran your notebook and found that mean and variance are getting updated while 'base_model.trainable = True'. There is typo in the guide of transfer learning. Raised PR for the same. Soon it will reflect once PR get merged. Thanks..!!
